polygeist/lib/Dialect/Polygeist/Transforms/RaiseToAffine.cpp

```diff
@@ -113,7 +113,7 @@ struct ForOpRaising : public OpRewritePattern<scf::ForOp> {
                    << "Failed: more than one lower/upper bounds\n");
         return failure();
       }
-      ubs[0] = rewriter.create<DivUIOp>(
+      ubs[0] = rewriter.create<CeilDivUIOp>(
```

Comment:
How do we make sure that we do not iterate too much when division is not exact?

Reply:
In both cases, the number of iterations will be `ceildiv(ub - lb, steps)`. In fact, a similar transformation is found in `affine` transformation code used to normalize affine loops ([see](https://github.com/intel/llvm/blob/3b45d74430e6c1db86be55871eebd8d25282e152/mlir/lib/Dialect/Affine/Utils/Utils.cpp#L553)).

Reply:
But if `lb=0`, `ub=9` and `steps=4`, with `ceildiv`, we're iterating three times, aren't we? Which is more than the original loop.

Reply:
In that case, we'd have:
- Original loop: `i = 0, 4, 8` (3 iterations)
- Affine loop: `i = 0, 1, 2` (3 iterrations)

As `ceildiv(9 - 0, 4) = ceildiv(9, 4) = 3`

Reply:
Got myself confused there, sorry.